conda_forge_tick/migrators/core.py

```diff
@@ -443,6 +445,51 @@ def __init__(
         self.top_level = top_level or set()
         self.cycles = set(chain.from_iterable(cycles or []))
 
+    def all_predecessors_issued_and_stale(self, attrs: "AttrsTypedDict") -> bool:
+        # Check if all upstreams have been issue and are stale
+        for node in self.graph.predecessors(attrs["feedstock_name"]):
+            try:
+                payload = self.graph.nodes[node]["payload"]
+            except KeyError as e:
+                print(node)
+                raise e
+
+            # we don't need to look at archived feedstocks
+            # they are always "migrated"
+            if payload.get("archived", False):
+                continue
+
+            muid = frozen_to_json_friendly(self.migrator_uid(payload))
+            pr_muids = _sanitized_muids(payload.get("PRed", []))
+            if muid not in pr_muids:
+                # not yet issued
+                return False
+            else:
+                # issued so check timestamp
+                pr_index = pr_muids.index(muid)
+                ts = (
+                    payload.get("PRed", [])[pr_index]
+                    .get("PR", {})
+                    .get("updated_at", None)
+                )
+                state = payload.get("PR", {"state": "open"}).get("state", "")
+                if ts is not None and state == "open":
+                    now = datetime.datetime.now(datetime.timezone.utc)
+                    ts = dateutil.parser.parse(ts)
+                    if now - ts < datetime.timedelta(days=30):
```

Comment:
Should we get the timedelta from the migration yaml itself? (And default to some default value?)

Reply:
The time delta is for the PRs from the migration, not when the migrator was started. We want to close when all PRs have been issued and every PR is at least 30 days old.

Reply:
Sorry. I meant more should this be a configurable field in the migration yaml. For instance some migrations may want to be longer/shorter on that time. It isn't a desperately needed feature though so we can punt if you'd like.

Reply:
Let's punt I think. 30 days is a fail-safe really and this only applies to migrations which have issued all PRs.